File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        subs = ""
        l = 0
        for char in s:
            index = subs.find(char)
            if index >= 0:
                subs = subs[index+1:]
                subs += char
            else:
                subs += char
                if len(subs) > l:
                    l += 1
                    
        return l

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int answer = 0;
        String longest = "";
        for (int i = 0; i < s.length(); i++){
            String ch = s.substring(i, i+1);
            if (longest.contains(ch)){
                int pos = longest.indexOf(ch);
                longest = longest.substring(pos+1) + ch;
            }
            else {
                longest += ch;
            }

            if (answer < longest.length()){
                answer = longest.length();
            }
        }

        return answer;
    }
}